esp/services/ws_packageprocess/ws_packageprocessService.cpp

```diff
@@ -1321,11 +1321,8 @@ bool CWsPackageProcessEx::onGetPackageMapSelectOptions(IEspContext &context, IEs
             ForEachItemIn(c, clusters)
             {
                 IConstWUClusterInfo &cluster = clusters.item(c);
-#ifndef _CONTAINERIZED
-                if (cluster.getPlatform() == RoxieCluster)
-#else
-                if ((cluster.getPlatform() == RoxieCluster) && cluster.isQueriesOnly())
-#endif
+                if ((cluster.getPlatform() == RoxieCluster) && ((cluster.getRoxieTargetType() == RTTQueries)
```

Comment:
Simpler if bitfield  can be used (and/or a helper function used to hide the details)

```
if ((cluster.getPlatform() == RoxieCluster) && (cluster.getRoxieTargetType() & RTTQueries))
```

or

```
if (cluster.hasDeployedQueries())
```

Reply:
Will add a canDeployQueries().